src/main/scala/introprog/PixelWindow.scala

```diff
@@ -290,6 +303,29 @@ class PixelWindow(
     }
   }
 
+  
+  /** Draw `image` at `(x, y)` scaled to `(width, height)`. */
+  def drawImage(
+    image: BufferedImage,
+    x: Int,
+    y: Int,
+    width: Int,
+    height: Int
+  ) = {
+    canvas.withGraphics { g =>
```

Comment:
g is not used more than once, so better use `_.drawImage(...` instead. Also use ( instead of { here.